core/server/src/main/java/alluxio/master/file/FileSystemMaster.java

```diff
@@ -2019,43 +2020,45 @@ private void mountFromEntry(AddMountPointEntry entry)
    * @throws InvalidPathException if an invalid path is encountered
    * @throws IOException if an I/O exception occurs
    */
-  private void mountInternal(LockedInodePath inodePath, AlluxioURI ufsPath, MountOptions options)
+  private void mountInternal(LockedInodePath inodePath, AlluxioURI ufsPath, boolean replayed,
+      MountOptions options)
       throws FileAlreadyExistsException, InvalidPathException, IOException {
     AlluxioURI alluxioPath = inodePath.getUri();
-    // Check that the ufsPath exists and is a directory
-    UnderFileSystem ufs = UnderFileSystem.get(ufsPath.toString(), MasterContext.getConf());
-    ufs.setProperties(options.getProperties());
-    if (!ufs.exists(ufsPath.toString())) {
-      throw new IOException(ExceptionMessage.UFS_PATH_DOES_NOT_EXIST.getMessage(ufsPath.getPath()));
-    }
-    if (ufs.isFile(ufsPath.toString())) {
-      throw new IOException(ExceptionMessage.PATH_MUST_BE_DIRECTORY.getMessage(ufsPath.getPath()));
-    }
-    // Check that the alluxioPath we're creating doesn't shadow a path in the default UFS
-    String defaultUfsPath = MasterContext.getConf().get(Constants.UNDERFS_ADDRESS);
-    UnderFileSystem defaultUfs = UnderFileSystem.get(defaultUfsPath, MasterContext.getConf());
-    if (defaultUfs.exists(PathUtils.concatPath(defaultUfsPath, alluxioPath.getPath()))) {
-      throw new IOException(
-          ExceptionMessage.MOUNT_PATH_SHADOWS_DEFAULT_UFS.getMessage(alluxioPath));
-    }
-    // This should check that we are not mounting a prefix of an existing mount, and that no
-    // existing mount is a prefix of this mount.
-    mMountTable.add(alluxioPath, ufsPath, options);
 
-    try {
+    if (!replayed) {
+      // Check that the ufsPath exists and is a directory
+      UnderFileSystem ufs = UnderFileSystem.get(ufsPath.toString(), MasterContext.getConf());
+      ufs.setProperties(options.getProperties());
+      if (!ufs.exists(ufsPath.toString())) {
+        throw new IOException(
+            ExceptionMessage.UFS_PATH_DOES_NOT_EXIST.getMessage(ufsPath.getPath()));
+      }
+      if (ufs.isFile(ufsPath.toString())) {
+        throw new IOException(
+            ExceptionMessage.PATH_MUST_BE_DIRECTORY.getMessage(ufsPath.getPath()));
+      }
+      // Check that the alluxioPath we're creating doesn't shadow a path in the default UFS
+      String defaultUfsPath = MasterContext.getConf().get(Constants.UNDERFS_ADDRESS);
+      UnderFileSystem defaultUfs = UnderFileSystem.get(defaultUfsPath, MasterContext.getConf());
+      if (defaultUfs.exists(PathUtils.concatPath(defaultUfsPath, alluxioPath.getPath()))) {
+        throw new IOException(
+            ExceptionMessage.MOUNT_PATH_SHADOWS_DEFAULT_UFS.getMessage(alluxioPath));
+      }
+
       // Configure the ufs properties, and update the mount options with the configured properties.
       ufs.configureProperties();
       options.setProperties(ufs.getProperties());
```

Comment:
no, the addition has been moved after this operation so there is no need to delete the mount point if the ufs operation throws an exception